File: shoring/desk/src/main/java/com/panopset/marin/oldswpkg/games/blackjack/LayoutDealer.java
package com.panopset.marin.oldswpkg.games.blackjack;

import com.panopset.compat.Tile;
import com.panopset.blackjackEngine.CycleSnapshot;

public class LayoutDealer extends Layout {

    public LayoutDealer(Tile t, int[] cardDim, CycleSnapshot cs) {
        super(t, cardDim, cs);
    }


    private Integer nextDealerX;

    public Integer getNextDealerX() {
        if (nextDealerX == null) {
            nextDealerX = t.getLeft();
        } else {
            nextDealerX = nextDealerX + cardSpacer;
        }
        return nextDealerX;
    }

}

File: code/shoring/desk/src/main/kotlin/com/panopset/marin/oldswpkg/games/blackjack/LayoutDealer.kt
package com.panopset.marin.oldswpkg.games.blackjack

import com.panopset.blackjackEngine.BlackjackGameState
import com.panopset.compat.Tile

class LayoutDealer(t: Tile?, cardDim: IntArray?, cs: BlackjackGameState?) : Layout(t!!, cardDim!!, cs!!) {
    var nextDealerX: Int? = null
        get() {
            field = if (field == null) {
                t.left
            } else {
                field!! + cardSpacer
            }
            return field
        }
        private set
}
